### PR TITLE
Update JavaDoc for Storage constructor

### LINKED ISSUE #67: Catch JSON user input errors when reading from data file
Check `Gson` API and implement some form of JSON format checking so that if a user edits the data file in such a way that it becomes corrupt on line `n`, `Storage` will read everything up until line `n` and discard everything after.

### PR DESCRIPTION
**Pull request by xenthm**
Fixes #67

### CODE REVIEW

**Comment by xenthm** [comment]
Input data file:
image


App output:
image

File: src/main/java/storage/Storage.java [hunk 1]
```diff
@@ -87,30 +95,33 @@ public static Storage getInstance() {
...
             return authorList;
         } catch (IOException e) {
-            logger.warning("Problems accessing file, data was not restored!" + e.getMessage());
-            System.out.println("Problems accessing file, data was not restored! Continuing with an empty list...");
+            logger.warning("Problems accessing file: " + e.getMessage());
+            System.out.println("Problems accessing file, data was not restored! Continuing with an empty list.");
+        } catch (JsonSyntaxException e) {
+            logger.warning("JSON from file is malformed: " + e.getMessage());
+            System.out.println(
+                    "JSON from file is malformed, data was not restored! Continuing with an empty list."
```
Comment (sarahchow03) [inline src/main/java/storage/Storage.java:108]: Good job catching the possibility of a corrupted file and handling the case for it!

File: src/test/java/storage/MangaDeserializerTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    void deserialize_validMangaObject_returnsManga() {
+        Author expectedAuthor = new Author("test");
+        Manga expectedManga = new Manga("manga1", expectedAuthor, "deadline1");
+
+        JsonObject mangaJson = new JsonObject();
+        mangaJson.addProperty("mangaName", "manga1");
+        mangaJson.addProperty("deadline", "deadline1");
```
Comment (sarahchow03) [inline src/test/java/storage/MangaDeserializerTest.java:97]: With the new addition of sales data, certain test cases may need to be updated to align with those changes.

**Review by sarahchow03 — APPROVED** [review]
LGTM!

**Pull request merged**